cmake/Modules/LibAddPlugin.cmake

```diff
@@ -45,7 +45,7 @@ function (add_plugintest testname)
 		restore_variable (${PLUGIN_NAME} ARG_COMPILE_DEFINITIONS)
 		restore_variable (${PLUGIN_NAME} ARG_INCLUDE_DIRECTORIES)
 		restore_variable (${PLUGIN_NAME} ARG_INCLUDE_SYSTEM_DIRECTORIES)
-		restore_variable (${PLUGIN_NAME} ARG_LINK_ELEKTRA)
+		restore_variable (${PLUGIN_NAME} ARG_LINK_ELEKTRA ALLOW_MATCH)
```

Comment:
@markus2330 the dbusrecv plugin [required io_uv for testing](https://github.com/ElektraInitiative/libelektra/pull/1825/files#diff-67c9a608911fa20894d47412d4fba5f2R28): While the dbusrecv plugin relies on the ElektraIoInterface, to test the code a I/O binding was required.

I relaxed the consistency checks so that `add_plugintest` allows additional libraries. The plugin libraries still need to be included.

Reply:
Would be better if we can avoid circular deps.

If we really would need such a hack, we should introduce LINK_LIBRARIES_CIRCULAR or similar which adds the ALLOW_MATCH (the name ALLOW_MATCH is not really good, too?)

Reply:
Why would the deps be circluar?
It allows to add additional libraries for plugin tests.

> If we really would need such a hack, we should introduce LINK_LIBRARIES_CIRCULAR or similar which adds the ALLOW_MATCH (the name ALLOW_MATCH is not really good, too?)

Okay. How about LINK_LIBRARIES_ADDITIONAL and "ALLOW_ADDITIONAL"?

Reply:
Ahh, I misunderstood what you are trying to achieve.

Yes, fresh names are a much better solution. What about UNITTEST_LINK_LIBRARIES? (ADDITIONAL is too generic, then it is unclear -- like it was for me -- that you only need it for the tests.)